azavea/raster-foundry
# Add ability to create tools from expressions

## Overview

This PR adds tool expression parsing, a somewhat experimental feature, into the front-end tool ecosystem.

The 'New Tool' button on the tool marketplace page now opens a modal in which a tool can be created.

There is _very_ limited validation on the tool definition itself other than it's existence (see #2226).

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

![image](https://user-images.githubusercontent.com/2442245/28078648-95ae9cf6-6633-11e7-8ed5-a5561db79ff6.png)

## Testing Instructions

 * Make some tools and verify that the requests going to the server reflect user inputs and reasonable defaults

Closes #2133
